### ISSUE TITLE
Lit element with closed shadow root still has a public property to access the shadow root

### ISSUE THREAD

**Issue by tomherni** [lit/lit#4020]
### Which package(s) are affected?

Lit Core (lit / lit-html / lit-element / reactive-element)

### Description

When I create a `LitElement` component with a closed shadow root, it still exposes a public property called `renderRoot` that returns the `#shadow-root` reference. Having this property as **public** seems to defeat the purpose of closed shadow roots.

The internal features of a closed shadow root are supposed to be inaccessible and unchangeable from JavaScript. Of course, for those who are creative enough, they will find a way to access them, but it won't be as simple as using a web component's **public** API.

I noticed in lit-element's changelog that `renderRoot` used to be `protected`, but was changed to `public readonly` in version 2.1.0 (dating way back to 2019).

_P.S. I would love for this issue to focus on Lit's design pattern and its 'support' for closed shadow roots, rather than delving into discussions about the downsides of closed shadow roots, the various ways to bypass them, how they cannot be used for security, etc._

### Reproduction

I created a [Lit Playground](https://lit.dev/playground/#project=W3sibmFtZSI6Im15LWVsZW1lbnQudHMiLCJjb250ZW50IjoiaW1wb3J0IHtodG1sLCBMaXRFbGVtZW50fSBmcm9tICdsaXQnO1xuaW1wb3J0IHtjdXN0b21FbGVtZW50fSBmcm9tICdsaXQvZGVjb3JhdG9ycy5qcyc7XG5cbmNvbnN0IHNoYWRvd1Jvb3RNb2RlOiBTaGFkb3dSb290TW9kZSA9ICdjbG9zZWQnO1xuY29uc3Qgc2hhZG93Um9vdE9wdGlvbnMgPSB7IC4uLkxpdEVsZW1lbnQuc2hhZG93Um9vdE9wdGlvbnMsIG1vZGU6IHNoYWRvd1Jvb3RNb2RlIH07XG5cbkBjdXN0b21FbGVtZW50KCdteS1lbGVtZW50JylcbmV4cG9ydCBjbGFzcyBNeUVsZW1lbnQgZXh0ZW5kcyBMaXRFbGVtZW50IHtcbiAgLy8gc3RhdGljIG92ZXJyaWRlIHNoYWRvd1Jvb3RPcHRpb25zID0gc2hhZG93Um9vdE9wdGlvbnM7XG5cbiAgcmVuZGVyKCkge1xuICAgIHJldHVybiBodG1sYDxwPkhlbGxvIHdvcmxkITwvcD5gO1xuICB9XG59XG4ifSx7Im5hbWUiOiJpbmRleC5odG1sIiwiY29udGVudCI6IjwhRE9DVFlQRSBodG1sPlxuPGhlYWQ-XG4gIDxzY3JpcHQgdHlwZT1cIm1vZHVsZVwiIHNyYz1cIi4vbXktZWxlbWVudC5qc1wiPjwvc2NyaXB0PlxuPC9oZWFkPlxuPGJvZHk-XG4gIDxteS1lbGVtZW50PjwvbXktZWxlbWVudD5cbiAgXG4gIDxzY3JpcHQ-XG4gICAgY29uc3QgbXlFbGVtZW50ID0gZG9jdW1lbnQucXVlcnlTZWxlY3RvcignbXktZWxlbWVudCcpO1xuXG4gICAgY3VzdG9tRWxlbWVudHMud2hlbkRlZmluZWQoJ215LWVsZW1lbnQnKS50aGVuKGFzeW5jICgpID0-IHtcbiAgICAgIGF3YWl0IG15RWxlbWVudC51cGRhdGVDb21wbGV0ZTtcbiAgICAgIGNvbnNvbGUubG9nKG15RWxlbWVudC5yZW5kZXJSb290LnF1ZXJ5U2VsZWN0b3IoJ3AnKSk7XG4gICAgfSk7XG4gIDwvc2NyaXB0PlxuPC9ib2R5PlxuIn0seyJuYW1lIjoicGFja2FnZS5qc29uIiwiY29udGVudCI6IntcbiAgXCJkZXBlbmRlbmNpZXNcIjoge1xuICAgIFwibGl0XCI6IFwiXjIuMC4wXCIsXG4gICAgXCJAbGl0L3JlYWN0aXZlLWVsZW1lbnRcIjogXCJeMS4wLjBcIixcbiAgICBcImxpdC1lbGVtZW50XCI6IFwiXjMuMC4wXCIsXG4gICAgXCJsaXQtaHRtbFwiOiBcIl4yLjAuMFwiXG4gIH1cbn0iLCJoaWRkZW4iOnRydWV9XQ). See the console log of a querySelector going into the component's closed shadow root using the component's public API.

### Workaround

I have not found a workaround.

### Is this a regression?

No or unsure. This never worked, or I haven't tried before.

### Affected versions

latest

### Browser/OS/Node environment

Browser: Chrome 114.0.5735.198
OS: macOS 13.4

**Comment by sorvell** [comment]
The current design and customization options make this hard to avoid exposing. In particular:
1. `renderRoot` is configurable, it doesn't have to be a shadowRoot.
2. it's a difficult thing to change since it's been public for awhile.

**Comment by justinfagnani** [comment]
Yeah, this is difficult for us to change right now. Decorators like `@query()` need access to the render root in order to work. We would need to find a way to pass them the render root, and it would be a lot of syntax overhead, ie the class would have to have it's own private field to pass around and it would require a closure argument to every decorator:

```ts
class MyElement extends LitElement {
  // We need #renderRoot here _and_ this.createRenderRoot() has to have a side-effect of setting
  // a private field in the base class to use in update()
  #renderRoot = this.createRenderRoot();

  @query('div', (el: MyElement) => el.#renderRoot)
  accessor #div: HTMLDivElement;

  @queryAll('li', (el: MyElement) => el.#renderRoot)
  accessor #items: HTMLDivElement;
}
```
